src/main/java/Duke.java

```diff
@@ -1,10 +1,119 @@
+    public static void userInput() {
+        Scanner scan = new Scanner(System.in);
+        System.out.println("Enter your wish: " + "\n" + line);
+        String input = scan.nextLine();
+        inputSort(input);
+    }
+
+    //Lists out all tasks stored and their statuses
+    public static void sayList() {
+        System.out.println(line);
+        for (int i = 0; i < taskCount; i++) {
+            System.out.println((i + 1) + ". " + t[i].toString()+ "\n");
+        }
+    }
+
+    //Marks a stored task as done
+    public static void sayDone(String input) {
+        String taskNumber = input.substring(input.lastIndexOf(" ") + 1);
+        int finalTaskNumber = Integer.parseInt(taskNumber) - 1;
+        t[finalTaskNumber].markAsDone();
+        System.out.println(line + "\n" + "Kudos! One less thing to stress about!\n");
+        System.out.println("  " + t[finalTaskNumber].toString() + "\n" + line);
+    }
+
+    //Adds a new todo task and prints it                                       //todo borrow book
+    public static void sayTodo(String input) {
+        int endIndex = input.length();
+        String taskName = input.substring(5, endIndex);
+        t[taskCount] = new Todo(taskName);
+        System.out.println(line + "\n");
+        System.out.println("Got it. I've added this task:\n");
+        System.out.println(t[taskCount].toString());
+        taskCount++;
+        System.out.println("\nNow you have " + taskCount + " tasks in the list.\n" + line);
+    }
+
+    //Adds a new deadline task and prints it
+    public static void sayDeadline(String input) {                               //deadline return book /by Sunday
+        int endIndex = input.lastIndexOf("/");
+        String taskName = input.substring(9, endIndex);
+        int endIndex2 = input.length();
+        String by = input.substring(endIndex + 4, endIndex2);
+        t[taskCount] = new Deadline(taskName, by);
+        System.out.println(line + "\n");
+        System.out.println("Got it. I've added this task:\n");
+        System.out.println(t[taskCount].toString());
+        taskCount++;
+        System.out.println("\nNow you have " + taskCount + " tasks in the list.\n" + line);
+    }
+
+    //Adds a new event task and prints it
+    public static void sayEvent(String input) {                                 //event project meeting /at Mon 2-4pm
+        int endIndex = input.lastIndexOf("/");
+        String taskName = input.substring(6, endIndex);
+        int endIndex2 = input.length();
+        String at = input.substring(endIndex + 4, endIndex2);
+        t[taskCount] = new Event(taskName, at);
+        System.out.println(line + "\n");
+        System.out.println("Got it. I've added this task:\n");
+        System.out.println(t[taskCount].toString());
+        taskCount++;
+        System.out.println("\nNow you have " + taskCount + " tasks in the list.\n" + line);
+    }
+
+    //Filters user inputs and pushes to different methods
```

Comment:
No coding violations found here, I like the use of comments at the top to give a brief explanation of what the function does